internal/ls/diagnostics.go

```diff
@@ -13,17 +13,24 @@ func (l *LanguageService) GetDocumentDiagnostics(ctx context.Context, documentUR
 	syntaxDiagnostics := program.GetSyntacticDiagnostics(ctx, file)
 	var lspDiagnostics []*lsproto.Diagnostic
 	if len(syntaxDiagnostics) != 0 {
-		lspDiagnostics = make([]*lsproto.Diagnostic, len(syntaxDiagnostics))
-		for i, diag := range syntaxDiagnostics {
-			lspDiagnostics[i] = toLSPDiagnostic(diag, l.converters)
+		lspDiagnostics = make([]*lsproto.Diagnostic, 0, len(syntaxDiagnostics))
+		for _, diag := range syntaxDiagnostics {
+			lspDiagnostics = append(lspDiagnostics, toLSPDiagnostic(diag, l.converters))
 		}
 	} else {
+		// !!! suggestion diagnostics?
+		// !!! In Strada, we consolidate bind/check errors into semantic diagnostics rather than
```

Comment:
Wait, we already consolidate them in `program.GetSemanticDiagnostics()`. Why are we not using it here?

Reply:
The code below asks for a checker and asks it for diagnostics directly, so nothing here is asking for them anymore.